pkg/recipes/driver/terraform.go

```diff
@@ -135,13 +135,28 @@ func (d *terraformDriver) Delete(ctx context.Context, opts DeleteOptions) error
 			logger.Info(fmt.Sprintf("Failed to cleanup Terraform execution directory %q. Err: %s", requestDirPath, err.Error()))
 		}
 	}()
-
+	// Add credential information to .gitconfig if module source is of type git.
+	if strings.HasPrefix(opts.Definition.TemplatePath, "git::") && !reflect.DeepEqual(opts.BaseOptions.Secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
```

Comment:
Can you please abstract this out in a separate function? That will make it easier to support for more private sources in the future.

Also, let's please update the comment to include context on why this is needed - to access private module sources.

Reply:
Yes I will add the comment, but can we skip the part of making a separate function ? because as part of supporting nested modules i will get rid of these setting and unsetting part of the code here.